internal/agent/agent.go

```diff
@@ -838,6 +847,25 @@ func (a *Agent) Run() (err error) {
 	}
 }
 
+// checkInstanceIDRetry will try to read the cloud instance ID until maxRetries is reached.
+func (a *Agent) checkInstanceIDRetry(maxRetries, backoffTime int) error {
+	var err error
+	for i := 0; i <= maxRetries; i++ {
+		if _, err = a.cloudHarvester.GetInstanceID(); err == nil {
+			return nil
+		}
+
```

Comment:
I would remove this if as it already returns the error if max retries is reached 👇 
```go
// checkInstanceIDRetry will try to read the cloud instance ID until maxRetries is reached.
func (a *Agent) checkInstanceIDRetry(maxRetries, backoffTime int) error {
	var err error
	for i := 0; i <= maxRetries; i++ {
		if _, err = a.cloudHarvester.GetInstanceID(); err == nil {
			return nil
		}

		alog.WithError(err).Debugf("Failed to get the instance ID, retrying in %d s.", backoffTime)
		time.Sleep(time.Duration(backoffTime) * time.Second)
	}

	return fmt.Errorf("failed to get an instance ID after %d retries: %w", maxRetries, err)
}
```

Reply:
That was my first approach, but then if the last loop attempt fails, it would log that it's going to retry but fail outright (for example, if `maxRetries` is 0 it should attempt getting the instance ID just one time before failing, but would print a `retrying in...`)

Reply:
I'll add some testing, thanks!

Reply:
>That was my first approach, but then if the last loop attempt fails, it would log that it's going to retry but fail outright (for example, if maxRetries is 0 it should attempt getting the instance ID just one time before failing, but would print a retrying in...)

Got it. On the other hand, that way it will print 2 messages at the end. What about?:
```go
	for i := 0; i <= maxRetries; i++ {
		if _, err = a.cloudHarvester.GetInstanceID(); err == nil {
			return nil
		}

		if i == maxRetries-1 {
			break
		} 
	        alog.WithError(err).Debugf("Failed to get the instance ID, retrying in %d s.", backoffTime)
	        time.Sleep(time.Duration(backoffTime) * time.Second)
	}
         return fmt.Errorf("failed to get an instance ID after %d retries: %w", maxRetries, err)
```

Reply:
Great! I'll go with that